### ISSUE TITLE
HTML extract breaks with SVG data URL

### ISSUE THREAD

**Issue by Arcanemagus** [stylelint/stylelint#1604]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

It seems that when the HTML extraction feature is enabled and a `url()` is specified as the source of a property that is actually a `data:` URI containing a SVG element, the extraction breaks leading to a `no-empty-source` rule violation.

> Which rule, if any, is this issue related to?

`no-empty-source` (Most likely only exposing the underlying bug)

> What CSS is needed to reproduce this issue?

```css
select {
  background-image: url("data:image/svg+xml;utf8,<svg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' width='24' height='24' viewBox='0 0 24 24'><path fill='#444' d='M7.406 7.828l4.594 4.594 4.594-4.594 1.406 1.406-6 6-6-6z'></path></svg>");
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "no-empty-source": true
  }
}
```

> Which version of stylelint are you using?

`6.8.0` (Node, CLI) and `6.9.0` (CLI)

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with:

* v6.8.0: `stylelint --extract .\foo.css`
* v6.9.0: `stylelint --extract true .\foo.css`

_Note: The `--extract` option needing a "buffer parameter" in v6.9.0 is filed as https://github.com/stylelint/stylelint/issues/1605_

Node API in Atom v1.9.0-beta0 (~Node.js v5.10.0) through `linter-stylelint@2.16.0`.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it relates to extraction of CSS from within HTML.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
foo.css
 1:1  ×  Unexpected empty source   no-empty-source
```
----

Note that I'm almost positive this is a bug in the HTML extraction code, as seems to be falsely triggering on the embedded SVG element. `linter-stylelint` will no longer be always setting the flag enabling that to true for all files on the next release, instead only enabling it for files where it should be ran, but it should still have not caused this issue in the first place.

**Comment by Arcanemagus** [comment]
Note, originally filed as https://github.com/AtomLinter/linter-stylelint/issues/224.

**Comment by evilebottnawi** [comment]
@Arcanemagus can you create repo with failes?

**Comment by jeddy3** [comment]
@Arcanemagus Thanks, as always, for the detailed write-up.

As this functionality is being pulled out to *processor* in `7.0.0`, do we want to put the effort into resolving this or do we just document the current limitation in the `no-empty-source` README? e.g.

"This rule is not compatible with the combination `-e` and `--extract` CLI flags or the `extractStyleTagsFromHtml` node API option, and HTML tags within CSS code e.g. the `<svg>` tag within `background: url("data:image/svg+xml;utf8,<svg>..</svg>");`"

**Comment by Arcanemagus** [comment]
Personally I would go with the "document it's being removed" approach, especially since the functionality is being removed in v7. The issue was reported as a bug to `linter-stylelint` so I forwarded it on to here so you guys were made aware of it 😉.

@evilebottnawi I'm not entirely sure what you are asking there, the files listed in the issue should reproduce this for you...

**Comment by jeddy3** [comment]
> Personally I would go with the "document it's being removed" approach

SGTM :)

**Comment by jeddy3** [comment]
No longer relevant now that `v7` is out.
